src/lando/headless_api/tests/test_automation_api.py

```diff
@@ -710,6 +714,189 @@ def raise_conflict(*args, **kwargs):
+        .strip()
+    )
+
+    subprocess.run(["git", "switch", main_branch], cwd=repo_path, check=True)
+
+    return main_commit, main_file, feature_commit, feature_file
+
+
+@pytest.mark.parametrize("strategy", [None, "ours", "theirs"])
+@pytest.mark.django_db
+def test_automation_job_merge_onto_success_git(
+    strategy,
+    repo_mc,
+    treestatusdouble,
+    git_automation_worker,
+    monkeypatch,
+    request,
+):
+    repo = repo_mc(SCM_TYPE_GIT)
+    scm = repo.scm
+    scm.push = mock.MagicMock()
+
+    # Create a repo with diverging history
+    main_commit, main_file, feature_commit, feature_file = (
+        _create_split_branches_for_merge(request, scm, repo.system_path)
+    )
+
+    job = AutomationJob.objects.create(
+        status=JobStatus.SUBMITTED,
+        requester_email="test@example.com",
+        target_repo=repo,
+    )
+    AutomationAction.objects.create(
+        job_id=job,
+        action_type="merge-onto",
+        data={
+            "action": "merge-onto",
+            "commit_message": f"Merge test with strategy {strategy}",
+            "strategy": strategy,
+            "target": feature_commit,
+        },
+        order=0,
+    )
+
+    git_automation_worker.worker_instance.applicable_repos.add(repo)
+
+    assert git_automation_worker.run_automation_job(job)
+    assert job.status == JobStatus.LANDED
+    assert scm.push.called
+    assert len(job.landed_commit_id) == 40
+
+
+@pytest.mark.parametrize("strategy", [None, "ours", "theirs"])
+@pytest.mark.django_db
+def test_automation_job_merge_onto_success_hg(
+    strategy,
+    repo_mc,
+    treestatusdouble,
+    hg_automation_worker,
+    monkeypatch,
+    request,
+):
+    repo = repo_mc(SCM_TYPE_HG)
+    scm = repo.scm
+    scm.push = mock.MagicMock()
+
+    repo_path = Path(repo.system_path)
+
+    # Create commits on a feature branch
+    _create_hg_commit(request, repo_path)
+    _create_hg_commit(request, repo_path)
+    _create_hg_commit(request, repo_path)
+    feature_commit = subprocess.run(
+        ["hg", "log", "-r", ".", "-T", "{node}"],
+        cwd=repo_path,
+        capture_output=True,
+        text=True,
+        check=True,
+    ).stdout.strip()
+
+    # Return to rev 0 and create mainline commits
+    subprocess.run(["hg", "update", "--clean", "-r", "0"], cwd=repo_path, check=True)
+    _create_hg_commit(request, repo_path)
+    _create_hg_commit(request, repo_path)
+    _create_hg_commit(request, repo_path)
+    main_commit = subprocess.run(
+        ["hg", "log", "-r", ".", "-T", "{node}"],
+        cwd=repo_path,
+        capture_output=True,
+        text=True,
+        check=True,
+    ).stdout.strip()
+
+    # Push changes to hg_server before running the automation job
+    subprocess.run(
+        ["hg", "push", "-r", "draft()", repo.push_path, "-f"], cwd=repo_path, check=True
+    )
+
+    subprocess.run(
+        ["hg", "update", "--clean", "-r", main_commit], cwd=repo.system_path, check=True
+    )
```

Comment:
Could we hide this in `_create_split_branches_for_merge`, and merge both success tests to be SCM-agnostic?

Reply:
We should do that as a follow up, it's failing on the hg side and not worth the trouble right now IMO. :)